lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -642,6 +680,330 @@ def get_fq_stats(self, reads_object, file_path):
+                                     file needs to be shared publicly 
+
+        params:
+        file_url: Google Drive download link
+        copy_file_path: output file saving path
+
+        """
+        # translate Google Drive URL for direct download
+        force_download_link_prefix = 'https://drive.google.com/uc?export=download&id='
+        file_id = file_url.partition('/d/')[-1].partition('/')[0]
+        force_download_link = force_download_link_prefix + file_id
+
+        self.log('Generating Google Drive direct download link\n from: %s\n to: %s' % (file_url, force_download_link))
+        self._download_direct_download_link(force_download_link, copy_file_path)
+
+    def _process_download(self, fwd, rev, reads_source, download_type, user_id):
+        """
+        _process_download: processing different type of downloads
+
+        fwd: forward shock_id if reads_source is 'shock'
+               forward url if reads_source is 'web'
+               forward file name in staging if reads_source is 'staging'
+        rev: reverse shock_id if reads_source is 'shock'
+               reverse url if reads_source is 'web'
+               reverse file name in staging if reads_source is 'staging'
+        reads_source: one of 'shock', 'web' or 'staging'
+        download_type: one of ['Direct Download', 'FTP', 'DropBox', 'Google Drive']
+        user_id: current token user
+        """
+
+        # return values
+        fwdpath = None
+        revpath = None
+        fwdname = None
+        revname = None
+  
+        if reads_source == 'shock':
+            # Grab files from Shock
+            fwdid = fwd
+            revid = rev
+            dfu = DataFileUtil(self.callback_url)
+            fileinput = [{'shock_id': fwdid,
+                          'file_path': self.scratch + '/fwd/',
+                          'unpack': 'uncompress'}]
+            if revid:
+                fileinput.append({'shock_id': revid,
+                                  'file_path': self.scratch + '/rev/',
+                                  'unpack': 'uncompress'})
+            self.log('downloading reads file(s) from Shock')
+            files = dfu.shock_to_file_mass(fileinput)
+            fwdpath = files[0]["file_path"]
+            fwdname = files[0]["node_file_name"]
+            if revid:
+                revpath = files[1]["file_path"]
+                revname = files[1]["node_file_name"]
+        elif reads_source == 'web':
+            # TODO: Tian move _download_web_file to DFU
+            fwdpath = self._download_web_file(fwd, download_type)
+            revpath = self._download_web_file(rev, download_type) if rev else None
+        elif reads_source == 'staging':
+            # TODO: Tian move _download_staging_file to DFU
+            fwdpath = self._download_staging_file(user_id, fwd)
+            revpath = self._download_staging_file(user_id, rev) if rev else None
+        elif reads_source == 'local':
+            pass
```

Comment:
set fwdpath and revpath to fwdid and revid here to maintain the same interface as the rest of the types (see below)

Also no need to declare *path above if you do this